las/client.py

```diff
@@ -89,303 +90,871 @@ def _create_signing_headers(self, path: str):
+        :type max_results: int
+        :param next_token: A unique token for each page, use the returned token to retrieve the next page.
+        :type next_token: str
+        :return: Users response from REST API
+        :rtype: dict
+
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        params = {
+            'maxResults': max_results,
+            'nextToken': next_token,
+        }
+        return self._make_request(requests.get, '/users', params=params)
 
     def get_user(self, user_id: str) -> dict:
-        """Get information about user, calls the GET /users/{user_id} endpoint.
+        """Get information about a specific user, calls the GET /users/{user_id} endpoint.
 
         >>> from las.client import BaseClient
         >>> client = BaseClient()
-        >>> client.get_user('me')
+        >>> client.get_user('<user_id>')
 
-        :param user_id: The user_id to get consent hash for
+        :param user_id: Id of the user
         :type user_id: str
-        :return: batch handle id and pre-signed upload url
+        :return: User response from REST API
         :rtype: dict
 
         :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
  :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
         """
-
         return self._make_request(requests.get, f'/users/{user_id}')
 
+    def delete_user(self, user_id: str) -> dict:
+        """Delete the user with the provided user_id, calls the DELETE /users/{userId} endpoint.
+
+        >>> from las.client import BaseClient
+        >>> client = BaseClient()
+        >>> client.delete_user('<user_id>')
+
+        :param user_id: Id of the user
+        :type user_id: str
+        :return: User response from REST API
+        :rtype: dict
+
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        return self._make_request(requests.delete, f'/users/{user_id}')
+
+    def create_workflow(self, specification: dict, name: str, description: Optional[str] = None,
+                        error_config: Optional[dict] = None) -> dict:
+        """Creates a new workflow, calls the POST /workflows endpoint.
+
+        >>> from las.client import BaseClient
+        >>> from pathlib import Path
+        >>> client = BaseClient()
+        >>> specification = {'language': 'ASL', 'version': '1.0.0', 'definition': {...}}
+        >>> error_config = {'email': '<error-recipient>'}
+        >>> client.create_workflow(specification, '<name>', '<description>', error_config)
+
+        :param specification: Specification of the workflow
```

Comment:
Would a link to https://states-language.net/spec.html make sense? + The limitations imposed by our API (can add later)